content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -1,48 +1,75 @@
+
 ### Definition of a Socket
-A socket is an **endpoint** of communication, acting as a connection between two computers on a network. It contains a socket number that the TCP layer can use to identify the application to which data will be transmitted. Additionally, an endpoint is made up of a port number and an IP address.
-There are two types of sockets
-1. Server socket - which waits for a request from clients
-2. Client socket - can be used to send and receive a request from clients.
-The client must know two information about the server:
-1.IP address of Server and 
+
+A Socket is a communication **endpoint** that serves as a link between two machines on a network. It has a socket number, which the TCP/IP layer can use to identify the application that receives the data. Additionally, a port number and an IP address are what make up an endpoint.
+
+Transmission Control Protocol (TCP) - This is one of the most important protocols of the Internet Protocols suite. It is a widely used protocol for data transmission in a network that operates in Client/Server point-to-point mode.
+
+Two categories of Sockets:
+
+1. A server socket awaits a request from a client.
+2. Client socket - utilized for sending and receiving requests from clients.
+
+The client has to know two things about the server:
+
+1.Server's IP address 
 2.Port number
->Port number 0-1024 are reserved for administrative purposes(e,g 21 for FTP,23 for Telnet,25 for email and 80 for HTTP).In our program, we will use port number 5000.
-Whenever we create a socket connection in Java, the following steps take place:
-1. Server creates a `ServerSocket` object, to indicate which port number our communication is to occur. If an I/O  error occurs, exception handling mechanisms are implemented; **Throws** an **I/OException** or a **try** **catch** block is used.
-2. The `accept()` function of the ServerSocket class is called by the server. This method waits for a client to connect to the server on a specific port before returning.
-3. A client builds a `Socket` object, giving the name of the server and the connecting port number as the server awaits.
-4. The `Socket` class constructor will try to connect the client to the server and port number provided. The client now has a `socket` object that if communication is established, may connect with the server.
-5. The `accept()` method on the server produces a reference to a new server socket that is connected to the client's socket.
+
+>The ports 0-1024 are for administrative use only (e,g 21 for FTP,23 for Telnet,25 for email, and 80 for HTTP). In our program, we'll be using port number 5000.
+
+When we make a socket connection in the Java programming language, we go through the following steps:
+
+- The Server constructs a `ServerSocket` object to specify the port number on which our conversation will occur. Exception handling methods are in use whenever an I/O error occurs. **Throws** an **I/OException** or a **try** **catch** block is used.
```

Comment:
> Exception handling methods are in use whenever an I/O error occurs. **Throws** an **I/OException** or a **try** **catch** block is used.

You still didn't remove the passive voice here but re-reading this, I think it's okay to keep this as is.